Clean up resource preview for practice quiz selection

rtibbles opened the pull request:
## Summary
* Don't show inaccurate headers for practice quizzes.
* Allow direct pressing of the "Select quiz" button when previewing the quiz.

## References
Fixes issues observed by @marcellamaki while reviewing another PR

## Reviewer guidance
Follow the select practice quiz workflow
Preview the quiz
Ensure all displayed buttons and information are relevant and functional.
Ensure quiz can be selected from the select quiz button and that it properly handles the selection action.

rtibbles committed:
Don't show inaccurate headers for practice quizzes.
Allow direct pressing of the "Select quiz" button when previewing the quiz.

marcellamaki commented:
@pcenov @radinamatic I've tagged this in as needs QA although I'm doing a first pass of manual QA now as I'm doing code review

marcellamaki reviewed (changes requested):
The code changes make sense to me, and the overall interaction seems to be working exactly as we discussed. The one lingering piece (which I didn't catch before) is that the snackbar continues to have the default 10 questions as the number added, rather than reflecting the actual number of questions. I've double checked, and the issue is specific to practice quizzes. I think it would be ideal to update this now as well if you don't mind.

https://github.com/user-attachments/assets/aa1c684e-cf06-421c-b950-b34b96da79ed

rtibbles committed:
Base number of questions in notification off number of questions added, explicitly.

rtibbles commented:
Thanks @marcellamaki I have updated - the current notification function was inferring the number of questions slightly magically, so I have changed it to take an explicit count instead and that seems to have resolved it.

Did a quick regression test on regular question adding (both auto select and manual) and no issues observed.

pcenov reviewed (approved):
The issue reported by Marcella is fixed now, everything else looks good too.
